packages/rust-dprint-plugin-typescript/src/main.rs

```diff
@@ -0,0 +1,189 @@
+                }
+            }
+            Err(e) => {
+                eprintln!("Error checking: {}", &file_path_str);
+                eprintln!("   {}", e);
+            }
+        }
+    }
+
+    if not_formatted_files.is_empty() {
+        Ok(())
+    } else {
+        let f = if not_formatted_files.len() == 1 {
+            "file"
+        } else {
+            "files"
+        };
+        Err(std::io::Error::new(
+            std::io::ErrorKind::Other,
+            format!("Found {} not formatted {}", not_formatted_files.len(), f,),
+        ))
+    }
+}
+
+fn format_source_files(config: Configuration, paths: Vec<PathBuf>) -> Result<(), std::io::Error> {
+    let mut not_formatted_files = vec![];
+
+    for file_path in paths {
+        let file_path_str = file_path.to_string_lossy();
+        let file_contents = fs::read_to_string(&file_path)?;
+
+        match dprint::format_text(&file_path_str, &file_contents, &config) {
+            Ok(None) => {
+                // nothing to format, pass
+            }
+            Ok(Some(formatted_text)) => {
+                if formatted_text != file_contents {
+                    println!("{}", file_path_str);
+                    fs::write(&file_path, formatted_text)?;
+                    not_formatted_files.push(file_path);
+                }
+            }
+            Err(e) => {
+                eprintln!("Error formatting: {}", &file_path_str);
+                eprintln!("   {}", e);
+            }
+        }
+    }
+
+    let f = if not_formatted_files.len() == 1 {
+        "file"
+    } else {
+        "files"
+    };
+    println!("Formatted {} {}", not_formatted_files.len(), f);
+    Ok(())
+}
+
+pub fn format(
+    config: dprint::configuration::Configuration,
+    target_files: Vec<PathBuf>,
+    check: bool,
+) -> Result<(), std::io::Error> {
+    if check {
+        check_source_files(config, target_files)?;
+    } else {
+        format_source_files(config, target_files)?;
+    }
+    Ok(())
+}
+
+fn create_cli_parser<'a, 'b>() -> clap::App<'a, 'b> {
+    App::new("dprint")
+        .about("Format source files")
+        .long_about(
+            "Auto-format JavaScript/TypeScript source code.
+  
+  dprint myfile1.ts myfile2.ts
+  
+  dprint --check myfile1.ts myfile2.ts
+  
+  dprint --config dprint.conf.json myfile1.ts myfile2.ts",
+        )
+        .arg(
+            Arg::with_name("check")
+                .long("check")
+                .help("Check if the source files are formatted.")
+                .takes_value(false),
+        )
+        .arg(
+            Arg::with_name("config")
+                .long("config")
+                .help("Path to JSON configuration file.")
+                .takes_value(true),
+        )
+        .arg(
+            Arg::with_name("files")
+                .help("List of file paths to format")
+                .takes_value(true)
+                .multiple(true)
+                .required(true),
+        )
+}
+
+fn main() {
```

Comment:
This is a huge nitpick (especially considering a lot of dprint is a mess at the moment 😅), but I'd like to make it anyway. When entering this file I was greeted with the `is_supported` function, but it would be nice if this function was at the top instead. That way it reads top down rather than bottom up. I know a lot of people are used to declare before use because of C, but I find it's kind of nice to read the high level code before the low level details 🙂

Anyway, not a big deal at all.

Reply:
Yeah, no problem 👍